core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -179,6 +180,11 @@ def remove_endpoints(self, endpoints_to_remove: Set[Endpoint]) -> bool:
             warn(f"Endpoints {missing_endpoints} not found. Aborting operation.")
             return False
 
+        protected_endpoints = set(filter(lambda endpoint: endpoint.protected, endpoints_to_remove))
```

Comment:
If protected is only used inside the if, it should be declared there.

Reply:
It's a typo on the if. It should be `if protected_endpoints`. Fixing